extensions/functions_string.yaml

```diff
@@ -133,6 +141,64 @@ scalar_functions:
           dotall:
             values: [ DOTALL_DISABLED, DOTALL_ENABLED ]
         return: "string"
+  -
+    name: regexp_match_substring_all
+    description: >-
+      Extract all substrings that match the given regular expression pattern. The regular expression
+      pattern should follow the International Components for Unicode implementation
+      (https://unicode-org.github.io/icu/userguide/strings/regexp.html). The number of characters
+      from the beginning of the string to begin starting to search for pattern matches can be
+      specified using the `position` argument. Specifying `1` means to search for matches
+      starting at the first character of the input string, `2` means the second character, and so
+      on. The `position` argument should be a positive non-zero integer. The regular
+      expression capture group can be specified using the `group` argument. Specifying `0`
+      will return substrings matching the full regular expression. Specifying `1` will return
+      substrings matching only the first capture group, and so on. The `group` argument should
+      be a non-negative integer.
```

Comment:
Why do I need to specify a `group` if it is `match_..._all`?  Wouldn't that return a value for each capture group?

Reply:
I understand the answer now.  However, can we clarify in the description that the "all" is referring to "occurrences" and not "groups"?  Maybe something like:

Extract all substrings that match the given regular expression pattern.  This will return a list of extracted strings with one value for each occurrence of a match.